### PR TITLE
Force trigger thread to be stopped on trigger removal, project disable, and project delete.

### PR DESCRIPTION
**Pull request by ggallen**

### CODE REVIEW

File: src/main/java/com/redhat/jenkins/plugins/ci/CIBuildTrigger.java [hunk 1]
```diff
@@ -173,7 +178,7 @@ private void startTriggerThread() {
 			}
 		}
 		if (job instanceof AbstractProject) {
```
Comment (scoheb) [inline src/main/java/com/redhat/jenkins/plugins/ci/CIBuildTrigger.java]: Do we need this if block anymore? I think workflow jobs support disable now..

Comment (ggallen) [reply src/main/java/com/redhat/jenkins/plugins/ci/CIBuildTrigger.java]: I have no idea.  I didn't change it because I wouldn't know.

Comment (ggallen) [reply src/main/java/com/redhat/jenkins/plugins/ci/CIBuildTrigger.java]: Scott, can we get this PR merged? If workflow jobs support disable now I think that is a different task.....

Comment (scoheb) [reply src/main/java/com/redhat/jenkins/plugins/ci/CIBuildTrigger.java]: I agree this can be in a different PR

**Pull request merged**